addon/components/frost-modal-binding.js

```diff
@@ -81,13 +143,40 @@ const FrostModalBinding = Component.extend(PropTypesMixin, {
       this.onClose()
     },
 
+    /**
+     * Handle confirmation in a promise-aware way.
+     * If onConfirm returns a promise, closeOnConfirm will wait for it to resolve successfully before closing the modal.
+     * @returns {Object} the return value of onConfirm.
```

Comment:
If `this.get('closeOnConfirm') !== true` this is not what gets returned

Reply:
Good point.  I think since there's no way to normally get the return value directly (it's just bound to the click handler), I had overlooked that. Thanks for pointing that out.